### PR TITLE
feat(TCOMP-572/salesforce):  Picklist fields of salesforce module are…

### PR DESCRIPTION
**Pull request by bdiouf**
… not returned

* Unitary test
* Do not skip PickList, address and location fields of salesforces modules

**What is the current behavior?** (You can also link to an open issue here)

**What is the new behavior?**

**Please check if the PR fulfills these requirements**

- [x] The commit message follows Talend standard
- [x] Tests for the changes have been added (for bug fixes / features)
- [ ] Docs have been added / updated (for bug fixes / features) ?
- [x] The code coverage on new code >75%
- [x] The new code does not introduce new technical issues (sonar / eslint)

**What kind of change does this PR introduce?**

- [ ] Bugfix
- [ ] Feature
- [ ] Code style update (formatting, local variables)
- [ ] Refactoring (no functional changes, no api changes)
- [ ] Build / CI related changes
- [ ] Other... Please describe:

**Does this PR introduce a breaking change?**

- [ ] Yes
- [ ] No

If this PR contains a breaking change, please describe the impact and migration path for existing applications: ...

**Other information**:

### CODE REVIEW

**Comment by bdiouf** [comment]
Within Dataprep we want to retrieve the PickList fields.
Could you please confirm that the code I have removed was not mandatory and does not break something in the Studio or elsewhere.

By the way I also removed the skip of location and address.

Thanks!

File: components/components-salesforce/components-salesforce-runtime/src/main/java/org/talend/components/salesforce/runtime/dataprep/SalesforceAvroRegistryString.java [hunk 1]
```diff
@@ -56,12 +55,6 @@ public Schema apply(DescribeSObjectResult t) {
     private Schema inferSchemaDescribeSObjectResult(DescribeSObjectResult in) {
         List<Schema.Field> fields = new ArrayList<>();
         for (Field field : in.getFields()) {
-            // filter the invalud columns for salesforce bulk query api
-            // not sure the picklist is a good filter for it
-            if (field.getType() == FieldType.address || field.getType() == FieldType.location
```
Comment (wwang-talend) [inline components/components-salesforce/components-salesforce-runtime/src/main/java/org/talend/components/salesforce/runtime/dataprep/SalesforceAvroRegistryString.java]: please don't filter the address and location as there is some limit for salesforce bulk api for compound type like address and location, you will meet some exception when read  the type value : 
 ```
stateMessage='InvalidBatch : Failed to process query: FUNCTIONALITY_NOT_ENABLED: Selecting compound data not supported in Bulk Query'
```

File: components/components-salesforce/components-salesforce-runtime/src/test/java/org/talend/components/salesforce/runtime/dataprep/SalesforceAvroRegistryStringTest.java [hunk 2]
```diff
@@ -0,0 +1,53 @@
...
+
+        assertThat(1, is(schema.getFields().size()));
+    }
+
+    @Test
+    public void testLocation() throws Exception {
+        DescribeSObjectResult describeSObjectResult = new DescribeSObjectResult();
+        Field location = new Field();
+        location.setName("location");
+        location.setType(FieldType.address);
```
Comment (wwang-talend) [inline components/components-salesforce/components-salesforce-runtime/src/test/java/org/talend/components/salesforce/runtime/dataprep/SalesforceAvroRegistryStringTest.java]: should be location type

**Comment by wwang-talend** [comment]
Hi Boubacar , have fixed it, please have a look if it meet your requirement.

**Comment by build-talend-common** [comment]
Pull request test result: [failure](https://ci-common.datapwn.com/job/components_pull-request_master/596/) [:x:](https://ci-common.datapwn.com/job/components_pull-request_master/596/)

**Comment by build-talend-common** [comment]
Pull request test result: [success](https://ci-common.datapwn.com/job/components_pull-request_master/597/) [:star:](https://ci-common.datapwn.com/job/components_pull-request_master/597/)

**Comment by bdiouf** [comment]
Thanks Wei, This is fine with me.
And again thans for your help :-)

**Pull request merged**